main.go

```diff
@@ -39,17 +40,30 @@ const (
 )
 
 func main() {
-	var filename string
-	var printVersion bool
-	var debug bool
-	var once bool
+
+	var (
+		filename     string
+		printVersion bool
+		debug        bool
+		once         bool
+		logfmt       string
+	)
 
 	flag.StringVar(&filename, "config", "", "path to config file")
 	flag.BoolVar(&printVersion, "version", false, "show version")
 	flag.BoolVar(&debug, "debug", false, "enable debugging mode")
 	flag.BoolVar(&once, "once", false, "run git2consul once and exit")
+	// allow switching logformat. Structured output helps with parsers
+	flag.StringVar(&logfmt, "logfmt", "text", "specify log format [text | json] ")
 	flag.Parse()
 
+	// Init checks
+	if len(filename) == 0 {
+		log.Error("No configuration file provided")
+		flag.Usage()
+		os.Exit(ExitCodeFlagError)
+	}
```

Comment:
Moving this check above the the "if printVersion" check on lines 71 to 74 causes the below bug.

```
/git2consul -version
2020/05/21 21:48:15 error No configuration file provided
Usage of ./git2consul:
  -config string
    	path to config file
  -debug
    	enable debugging mode
  -logfmt string
    	specify log format [text | json]  (default "text")
  -once
    	run git2consul once and exit
  -version
    	show version
```
```

Reply:
oops. Let me fix this and may be write a few test cases to cover these cases as well for future changes

Reply:
@seanmalloy thank you for spending time on this PR. I have fixed this issue and added two checks in e2e pkg. Looks like coverage is showing diff for auth.go but I have not made any changes there so may be it's a false positive